multisense_ros/cfg/multisense.cfg

```diff
@@ -2,297 +2,343 @@
-        gen.add("magnetometer_range", int_t, 0, "Magnetometer Range", 0, edit_method=m_range_enum);
-    #endif
-
-    clipping_enum = gen.enum([ gen.const("None", int_t, 0, "No Border Clip"),
-                               gen.const("Rectangular", int_t, 1, "Rectangular Border Clip"),
-                               gen.const("Circular", int_t, 2, "Circular Border Clip")],
-                              "Available border clipping options")
-
-    if "4000" in cfg.name:
-        clipping_max = 400.0
-    else:
-        clipping_max = 200.0
-    #endif
-
-    gen.add("border_clip_type", int_t, 0, "point cloud border clip type", 0, 0, 2, edit_method=clipping_enum)
-    gen.add("border_clip_value", double_t, 0, "point cloud border clip value", 0.0, 0.0, clipping_max)
-
-    gen.add("max_point_cloud_range", double_t, 0, "max point cloud range", 15.0, 0.0, 100.0)
-
-    gen.add("origin_from_camera_position_x_m", double_t, 0, "Origin from camera extrinsics transform x value (m)", 0.0, -100.0, 100.0)
-    gen.add("origin_from_camera_position_y_m", double_t, 0, "Origin from camera extrinsics transform y value (m)", 0.0, -100.0, 100.0)
-    gen.add("origin_from_camera_position_z_m", double_t, 0, "Origin from camera extrinsics transform z value (m)", 0.0, -100.0, 100.0)
-    gen.add("origin_from_camera_rotation_x_deg", double_t, 0, "Origin from camera extrinsics transform angle axis rotation x (deg)", 0.0, -180.0, 180.0)
-    gen.add("origin_from_camera_rotation_y_deg", double_t, 0, "Origin from camera extrinsics transform angle axis rotation y (deg)", 0.0, -180.0, 180.0)
-    gen.add("origin_from_camera_rotation_z_deg", double_t, 0, "Origin from camera extrinsics transform angle axis rotation z (deg)", 0.0, -180.0, 180.0)
-
-    gen.generate(PACKAGE, cfg.name, cfg.name)
+        if cfg.motor:
+            gen.add("motor_speed", double_t, 0, "MotorSpeed", 0.0, 0.0, 5.2);
+        #endif
+
+        gen.add("network_time_sync", bool_t, 0, "NetworkTimeSynchronization", True);
+
+        if cfg.name.find('sgm_AR0234') != -1:
+            gen.add("ptp_time_sync", bool_t, 0, "PTPTimeSynchronization", False);
+            trigger_source_enum = gen.enum([ gen.const("internal", int_t, 0, ""),
+                                            gen.const("ptp", int_t, 3, "")],
+                                            "Trigger sources");
+            gen.add("trigger_source", int_t, 0, "Trigger Source", 0, edit_method=trigger_source_enum)
+
+            gen.add("detail_disparity_profile", bool_t, 0, "DetailDisparityProfile", False);
+            gen.add("high_contrast_profile", bool_t, 0, "HighContrastProfile", False);
+            gen.add("show_roi_profile", bool_t, 0, "ShowRoiProfile", False);
+            gen.add("full_res_aux_profile", bool_t, 0, "FullResAuxProfile", False);
+
+            if feature == SupportedFeatures.GROUND_SURFACE:
+                gen.add("ground_surface_profile", bool_t, 0, "GroundSurfaceProfile", False);
+            #endif
+        #endif
```

Comment:
Here's where we generate a different config as we iterate over the supported features